perfkitbenchmarker/linux_benchmarks/speccpu2006_benchmark.py

```diff
@@ -12,120 +12,301 @@
-                     'mark metadata with partial=true. If unset, partial '
-                     'failures are treated as errors.')
+flags.DEFINE_enum(
+    'benchmark_subset', 'int', ['int', 'fp', 'all'],
+    'Used by the PKB speccpu2006 benchmark. Specifies a subset of SPEC CPU2006 '
+    'benchmarks to run.')
+flags.DEFINE_string(
+    'runspec_config', 'linux64-x64-gcc47.cfg',
+    'Used by the PKB speccpu2006 benchmark. Name of the cfg file to use as the '
+    'SPEC CPU2006 config file provided to the runspec binary via its --config '
+    'flag. If the benchmark is run using the cpu2006-1.2.iso file, then the '
+    'cfg file must be placed in the local PKB data directory and will be '
+    'copied to the remote machine prior to executing runspec. See README.md '
+    'for instructions if running with a repackaged cpu2006v1.2.tgz file.')
+flags.DEFINE_integer(
+    'runspec_iterations', 3,
+    'Used by the PKB speccpu2006 benchmark. The number of benchmark iterations '
+    'to execute, provided to the runspec binary via its --iterations flag.')
+flags.DEFINE_string(
+    'runspec_define', '',
+    'Used by the PKB speccpu2006 benchmark. Optional comma-separated list of '
+    'SYMBOL[=VALUE] preprocessor macros provided to the runspec binary via '
+    'repeated --define flags. Example: numa,smt,sse=SSE4.2')
+flags.DEFINE_boolean(
+    'runspec_enable_32bit', False,
+    'Used by the PKB speccpu2006 benchmark. If set, multilib packages will be '
+    'installed on the remote machine to enable use of 32-bit SPEC CPU2006 '
+    'binaries. This may be useful when running on memory-constrained instance '
+    'types (i.e. less than 2 GiB memory/core), where 64-bit execution may be '
+    'problematic.')
+flags.DEFINE_boolean(
+    'runspec_keep_partial_results', False,
+    'Used by the PKB speccpu2006 benchmark. If set, the benchmark will report '
+    'an aggregate score even if some of the SPEC CPU2006 component tests '
+    'failed with status "NR". Available results will be saved, and PKB samples '
+    'will be marked with a metadata value of partial=true. If unset, partial '
+    'failures are treated as errors.')
 
 BENCHMARK_NAME = 'speccpu2006'
 BENCHMARK_CONFIG = """
 speccpu2006:
-  description: Run Spec CPU2006
+  description: Runs SPEC CPU2006
   vm_groups:
     default:
       vm_spec: *default_single_core
       disk_spec: *default_500_gb
 """
 
-SPECCPU2006_TAR = 'cpu2006v1.2.tgz'
-SPECCPU2006_DIR = 'cpu2006'
+_BENCHMARK_SPECIFIC_VM_STATE_ATTR = 'speccpu2006_vm_state'
+_MOUNT_DIR = 'cpu2006_mnt'
+_SPECCPU2006_DIR = 'cpu2006'
+_SPECCPU2006_ISO = 'cpu2006-1.2.iso'
+_SPECCPU2006_TAR = 'cpu2006v1.2.tgz'
+_TAR_REQUIRED_MEMBERS = 'cpu2006', 'cpu2006/bin/runspec'
 
 
 def GetConfig(user_config):
   return configs.LoadConfig(BENCHMARK_CONFIG, user_config, BENCHMARK_NAME)
 
 
-def CheckPrerequisites():
-  """Verifies that the required resources are present.
+class SpecCpu2006ConfigSpec(benchmark_config_spec.BenchmarkConfigSpec):
```

Comment:
Can you explain why creating a BenchmarkConfigSpec subclass is preferable over just using CheckPrerequisites? CheckPrerequisites seems more appropriate to me, but I might be missing something